go/v1beta1/storage/memstore.go

```diff
@@ -0,0 +1,374 @@
+	nName := name.FormatNote(pID, nID)
+	m.Lock()
+	defer m.Unlock()
+	if _, ok := m.notesByName[nName]; !ok {
+		return nil, errors.Newf(codes.NotFound, "Note with name %q does not Exist", nName)
+	}
+
+	// TODO(#312): implement the update operation
+	n.UpdateTime = ptypes.TimestampNow()
+	m.notesByName[nName] = n
+	return n, nil
+}
+
+// DeleteNote deletes the specified note in memstore.
+func (m *memStore) DeleteNote(ctx context.Context, pID, nID string) error {
+	nName := name.FormatNote(pID, nID)
+	m.Lock()
+	defer m.Unlock()
+	if _, ok := m.notesByName[nName]; !ok {
+		return errors.Newf(codes.NotFound, "Note with name %q does not Exist", nName)
+	}
+	delete(m.notesByName, nName)
+	return nil
+}
+
+// GetOccurrenceNote gets the note for the specified occurrence from memstore.
+func (m *memStore) GetOccurrenceNote(ctx context.Context, pID, oID string) (*gpb.Note, error) {
+	oName := name.FormatOccurrence(pID, oID)
+	m.RLock()
+	defer m.RUnlock()
+	o, ok := m.occurrencesByName[oName]
+	if !ok {
+		return nil, errors.Newf(codes.NotFound, "Occurrence with name %q does not Exist", oName)
+	}
+	n, ok := m.notesByName[o.NoteName]
+	if !ok {
+		return nil, errors.Newf(codes.NotFound, "Note with name %q does not Exist", o.NoteName)
+	}
+	return n, nil
+}
+
+// ListNoteOccurrences returns up to pageSize number of occcurrences on the particular note (nID)
+// for this project (pID) projects beginning at pageToken (or from start if pageToken is the empty string).
+func (m *memStore) ListNoteOccurrences(ctx context.Context, pID, nID, filter, pageToken string, pageSize int32) ([]*gpb.Occurrence, string, error) {
+	// TODO: use filters
+	m.RLock()
+	defer m.RUnlock()
+	// Verify that note exists
+	if _, err := m.GetNote(ctx, pID, nID); err != nil {
+		return nil, "", err
+	}
+	nName := name.FormatNote(pID, nID)
+	os := []*gpb.Occurrence{}
+	for _, o := range m.occurrencesByName {
+		if o.NoteName == nName {
+			os = append(os, o)
+		}
+	}
+	sort.Slice(os, func(i, j int) bool {
+		return os[i].Name < os[j].Name
+	})
+	startPos := parsePageToken(pageToken, 0)
+	endPos := min(startPos+int(pageSize), len(os))
+	return os[startPos:endPos], nextPageToken(endPos, len(os)), nil
+}
+
+// GetVulnerabilityOccurrencesSummary gets a summary of vulnerability occurrences from storage.
+func (m *memStore) GetVulnerabilityOccurrencesSummary(ctx context.Context, projectID, filter string) (*gpb.VulnerabilityOccurrencesSummary, error) {
+	return &gpb.VulnerabilityOccurrencesSummary{}, nil
+}
+
+// Parses the page token to an int. Returns defaultValue if parsing fails
+func parsePageToken(pageToken string, defaultValue int) int {
+	if pageToken == "" {
+		return defaultValue
+	}
+	parsed, err := strconv.Atoi(pageToken)
+	if err != nil {
+		return defaultValue
+	}
+	return parsed
+}
+
+// Returns the smallest of a and b
+func min(a, b int) int {
+	if a < b {
+		return a
+	} else {
```

Comment:
You don't need the else clause, since you're exiting early.

Reply:
Good point, updated.